avm/res/network/dnsresolver/inbound-endpoint/main.bicep

```
@@ -0,0 +1,56 @@
+metadata name = 'DNS Resolver Inbound Endpoint'
+metadata description = 'This module deploys a DNS Resolver Inbound Endpoint.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Private DNS Resolver.')
+@minLength(1)
+param dnsResolverName string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Required. The subnet ID of the inbound endpoint.')
+param subnetId string
+
+@description('Optional. The private IP address of the inbound endpoint.')
+param privateIpAddress string?
+
+@description('Optional. The private IP allocation method of the inbound endpoint.')
+param privateIpAllocationMethod string = 'Dynamic'
+
+@description('Required. The name of the inbound endpoint.')
+param name string
+
+resource dnsResolver 'Microsoft.Network/dnsResolvers@2022-07-01' existing = {
+  name: dnsResolverName
+}
+
+resource dnsResolver_inboundEndpoint 'Microsoft.Network/dnsResolvers/inboundEndpoints@2022-07-01' = {
```

```suggestion
resource inboundEndpoint 'Microsoft.Network/dnsResolvers/inboundEndpoints@2022-07-01' = {
```

Please also update references & files. 
We used to only use the `_` for cases were in this case `dnsResolver` would be deployed an be the main resource. Here however it is the `inboundEndpoint` throwing the party